btr-web/btr-main-app/components/individual-person/tax-info/TaxNumber.vue

```diff
@@ -0,0 +1,115 @@
+<template>
+  <div class="flex flex-col py-5">
+    <div class="flex mb-2 py-1">
+      <URadio
+        id="taxNumberRadioButton"
+        v-model="selectedButton"
+        :value="HAS_TAX_NUMBER"
+        class="mt-2"
+        @change="handleRadioButtonChange(HAS_TAX_NUMBER)"
+      />
+      <UFormGroup :name="name" class="ml-5">
+        <UInput
+          v-model="taxNumber"
+          type="text"
+          variant="bcGov"
+          :placeholder="$t('placeholders.taxNumber')"
+          class="w-80"
+          @blur="formatInput"
+          @focus="handleInputFocus"
+        />
+      </UFormGroup>
+    </div>
+    <div class="flex items-center mb-2 py-2">
+      <URadio
+        id="noTaxNumberRadioButton"
+        v-model="selectedButton"
+        :value="NO_TAX_NUMBER"
+        @change="handleRadioButtonChange(NO_TAX_NUMBER)"
+      />
+      <label class="ml-5"> {{ $t('labels.noTaxNumberLabel') }} </label>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+const HAS_TAX_NUMBER = 'hasTaxNumber'
+const NO_TAX_NUMBER = 'noTaxNumber'
+
+defineProps({
+  id: { type: String, required: true },
+  name: { type: String, default: 'name' },
+  modelValue: {
+    type: Object,
+    default: () => ({
+      hasTaxNumber: undefined,
+      taxNumber: undefined
+    })
+  }
+})
+
+const emit = defineEmits<{
+  'update:modelValue': [value: { hasTaxNumber: boolean; taxNumber: string | undefined }]
+}>()
+
+// Indicate which radio button is selected.
+const selectedButton = ref('')
+
+// The tax number input value
+const taxNumber = ref('')
+
+watch(taxNumber, (newTaxNumber) => {
+  // when the user starts typing, the radio HAS_TAX_NUMBER button should be selected
+  if (newTaxNumber !== '') {
+    selectedButton.value = HAS_TAX_NUMBER
+    emit('update:modelValue', { hasTaxNumber: true, taxNumber: newTaxNumber })
+  }
+})
+
+const handleRadioButtonChange = (value) => {
+  selectedButton.value = value
+  if (value === NO_TAX_NUMBER) {
+    taxNumber.value = ''
+    emit('update:modelValue', { hasTaxNumber: false, taxNumber: undefined })
+  } else if (value === HAS_TAX_NUMBER) {
```

Comment:
why do you have this extra `else if`?

Can you do this instead?:
```
if (value === NO_TAX_NUMBER) {
    taxNumber.value = ''
}
emit('update:modelValue', { hasTaxNumber: true, taxNumber: taxNumber.value })
```

Or is there a reason you set it to 'undefined' ?

Reply:
When NO_TAX_NUMBER radio button is selected, the taxNumber field in the modelValue should be undefined. If it is an empty string. Zod will treat it as an invalid tax number and raise error. 

The "else-if" should be replaced with "else". So this code would work:

` if (value === NO_TAX_NUMBER) {
    taxNumber.value = ''
    emit('update:modelValue', { hasTaxNumber: false, taxNumber: undefined })
  } else {
    emit('update:modelValue', { hasTaxNumber: true, taxNumber: taxNumber.value })
  }`